### ISSUE TITLE
Dapr v1.11 runtime service invocation adds a default content type in dotnet sdk

### ISSUE THREAD

**Issue by yousamasham** [dapr/dapr#7861]
I am trying to invoke a python service from the dotnet sdk and I noticed that the `CreateInvokeMethodRequest` method returns an `HttpRequestMessage` instance that already has the Content-type header set to `application/json`. I am using Dapr runtime v1.11 and notice the same behaviour happens with Dapr runtime v1.13 as well.
![image](https://github.com/dapr/dapr/assets/45107433/914bbf96-b965-4c53-8eb0-34c270818533)

I thought this default content type was deprecated from Dapr runtime v1.7 and removed in Dapr runtme v1.9. So how come it is still being set to default to `application/json`?

To add on, when the request reaches the python service the json body is unable to be parsed and I have to add the following line of code to ensure that it is successfully able to parse it.
![image](https://github.com/dapr/dapr/assets/45107433/31d3156a-0c1f-44e0-a3c0-6ce858ea76bf)

Which is odd because the content type should already be set (see first image)

Thanks in advance,
Yousam

**Comment by yaron2** [comment]
The default content type was removed in Dapr, but the Dotnet SDK sets this for the method you're calling still. The recommendation for service invocation is to use the provided HTTP client and create an HTTP request yourself.

**Comment by yousamasham** [comment]
Thanks @yaron2 thanks for your quick response. I did some more investigation and testing and noticed something strange.

I noticed is that my dotnet to python service invocation breaks  when upgrading from runtime v1.11 to v1.13 if the request content from dotnet is of type `System.Net.Http.Json.JsonContent`. This was working fine in Dapr runtime v1.11 however it is breaking with v1.13.
I took a look through all the runtime changes for v1.13 but I could not find the change that was causing this break. Wondering if you would know/guide me towards a change that you suspect caused this so I can ensure that my program does not break unexpectedly with future version upgrades of the Dapr runtime?
